flink-core/src/main/java/org/apache/flink/api/common/state/StateTtlConfig.java

```diff
@@ -437,18 +465,43 @@ public boolean runCleanupForEveryRecord() {
                 DEFAULT_ROCKSDB_COMPACT_FILTER_CLEANUP_STRATEGY =
                         new RocksdbCompactFilterCleanupStrategy(1000L);
 
+        /**
+         * Default value lets RocksDB control this feature as needed. For now, RocksDB will change
+         * this value to 30 days (i.e 30 * 24 * 60 * 60) so that every file goes through the
+         * compaction process at least once every 30 days if not compacted sooner.
+         */
+        static final long DEFAULT_PERIODIC_COMPACTION_SECONDS = 0xfffffffffffffffeL;
```

Comment:
Why we enable this feature by default?
If enabled, I think a default value with 31 days looks better.

Reply:
Just as you commented in the frocksdb:
`Default: 0xfffffffffffffffe (allow RocksDB to auto-tune) For now, RocksDB will change this value to 30 days
  (i.e 30 * 24 * 60 * 60) so that every file goes through the compaction process at least once every 30 days if not compacted sooner.`
 IIUC, it's the default value of this feature in RocksDB ?
 Or Do we need to introduce a specific default value for flink meaning not setting this option ? WDYT ?

Reply:
I tried to run application without setting this config, and it shows that `Options.periodic_compaction_seconds: 0`, this is different from the docs.

Reply:
Oh, I have fixed it.
It should run normally now.